### PR TITLE
Add custom layout support to MAGI

### PR DESCRIPTION
**Pull request by connorgr**
Uses jQuery UI and vanilla CSS to provide new support for repositionable and resizable MAGI charts. Additionally, the visualizations are now responsive on window resize, which has been a bug since launch.

### CODE REVIEW

File: views/layout.jade [hunk 1]
```diff
@@ -6,6 +6,7 @@ html
     title MAGI
     link(rel='stylesheet', href='/components/bootstrap/dist/css/bootstrap.min.css')
     link(rel='stylesheet', href='/components/fontawesome/css/font-awesome.min.css')
+    link(rel="stylesheet" href="//code.jquery.com/ui/1.12.1/themes/base/jquery-ui.css")
```
Comment (mdml) [inline views/layout.jade]: Why not in Bower?

File: public/js/view.js [hunk 2]
```diff
@@ -347,11 +255,134 @@ function view(){
...
+
+		// And add tooltips
+		var intervals = cnas.selectAll("g.intervals"),
+			cnaTooltips = [];
+		intervals.classed("gd3-tipobj", true);
+		intervals.each(function(d) {
+			cnaTooltips.push([
+				{ type: 'link', href: '/sampleView?sample=' + d.sample, body: 'Sample: ' + d.sample },
+				{ type: 'text', text: 'Dataset: ' + d.dataset },
+				{ type: 'text', text: 'Type: ' + mutationToName(d.ty) },
```
Comment (mdml) [inline public/js/view.js]: This is broken. Perhaps should be `VIEW_UTIL.mutationToName`?

File: public/js/view.js [hunk 3]
```diff
@@ -36,213 +36,118 @@ var abbrToCancer = data.abbrToCancer,
...
 		else if (m == "del") return "Deletion";
 		else if (m == "expression") return "Expression";
 		else if (m == "methylation") return "Methylation";
 		else return "Other";
-	}
-
-	function pubmedLink(_id){
-		if (_id.toLowerCase().slice(0, 3) == 'pmc'){
+  },
+  pubmedLink: function(_id) {
```
Comment (mdml) [inline public/js/view.js:110]: References to this function are also broken.

File: public/js/view.js [hunk 4]
```diff
@@ -36,213 +36,118 @@ var abbrToCancer = data.abbrToCancer,
...
-					 {name: "cnas", el: cnasElement},
-					 {name: "heatmap", el: heatmapElement} ];
-
-
-	// Functions for giving names from our abbreviations
-	// for cancer types and mutations
-	function cancerToName(c){
-		if (cancerToAbbr[c]) return cancerToAbbr[c].toUpperCase();
+var VIEW_UTIL = {
+  cancerToName: function(c) {
```
Comment (mdml) [inline public/js/view.js:87]: References to this function are also broken.

**Review by mdml — CHANGES_REQUESTED** [review]
## Overall
I tested by updating [MAGI](http://magi.brown.edu) to use this branch, since none of these changes affect the server-side app or database. Very nice work, and very useful!

@braphael8: please feel free to check out the live version to make sure I didn't miss any bugs or missing features.

## Questions
Is it easy to make the views vertically re-sizable? Seems like an obvious enhancement if not. Would also be very useful.

## Bugs
### Breaking
I did find some minor but breaking bugs, all related to the `VIEW_UTIL` functions. Can you fix?

### Non-breaking
I also found some non-breaking bugs:

1. The ability to hide the network and transcript views appears to be broken. Here's a screenshot where only the network SVG is hidden, rather than the whole panel body.
screen shot 2017-04-26 at 8 18 05 am

2. The gradient legend for the network view is broken.
screen shot 2017-04-26 at 8 19 58 am

3. Not sure if this was already a problem, but the sort window is partially hidden.
screen shot 2017-04-26 at 8 17 48 am

**Comment by braphael8** [comment]
This looks great!  I didn't find any additional bugs beyond what Max reported above.

Being able to resize vertically would be extremely helpful; e.g. one might want to make the transcript view taller to show all mutations, rather than have to scroll.

File: views/layout.jade (hunk 1 above)
Comment (connorgr) [reply views/layout.jade]: fixed in b2c8466

File: public/js/view.js (hunk 4 above)
Comment (connorgr) [reply public/js/view.js:87]: fixed in 53d7ab3

File: public/js/view.js (hunk 3 above)
Comment (connorgr) [reply public/js/view.js:110]: fixed in 53d7ab3

File: public/js/view.js (hunk 2 above)
Comment (connorgr) [reply public/js/view.js]: fixed in 53d7ab3

**Comment by connorgr** [comment]
@mdml 

Network color bug fixed in cd3f096

Aberration matrix sort bug fixed in f40b071

The resizing bug isn't necessarily a bug since it's expected behavior of the grid layout. For each row the elements will only vertically resize to the smallest element. The same thing happens for all the visualizations if you resize them to be half-width and then hide. One fix would be to just completely hide the panel body instead of resizing it using collapse, but that's a pretty big change to how the front end works right now.

I just took a stab at adding vertical resizing for the network and transcript charts, which is in a3094d8.

**Review by mdml — APPROVED** [review]

**Pull request merged**